compiler/noirc_frontend/src/resolve_locations.rs

```diff
@@ -53,28 +57,49 @@ impl NodeInterner {
     /// For a given [Index] we return [Location] to which we resolved to
     /// We currently return None for features not yet implemented
     /// TODO(#3659): LSP goto def should error when Ident at Location could not resolve
-    fn resolve_location(&self, index: impl Into<Index>) -> Option<Location> {
+    fn resolve_location(
+        &self,
+        index: impl Into<Index>,
+        return_type_location_instead: bool,
+    ) -> Option<Location> {
```

Comment:
We can move the `if return_type_location_instead` check to the top of this function instead where we can get the type via `self.id_type(index)`. Since the check is at the very top of this function now, we can move it to a separate function as well to resolve a type from an index, then we can use that function to get the location of the type.

I think the "get type from index" function will be useful in implementing "hover" functionality later to show the type of a variable.

Reply:
addressed in
14202f1: chore: extract type resolution to top